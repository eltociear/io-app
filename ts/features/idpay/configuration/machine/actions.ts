import { IOToast } from "@pagopa/io-app-design-system";
import * as O from "fp-ts/lib/Option";
import { pipe } from "fp-ts/lib/function";
import I18n from "../../../../i18n";
import { useIONavigation } from "../../../../navigation/params/AppParamsList";
import { guardedNavigationAction } from "../../../../xstate/helpers/guardedNavigationAction";
import { IDPayDetailsRoutes } from "../../details/navigation";
import { IdPayConfigurationRoutes } from "../navigation/routes";
import { InitiativeFailure } from "../types/failure";
import * as Context from "./context";

const createActionsImplementation = (
  navigation: ReturnType<typeof useIONavigation>
) => {
  const navigateToConfigurationIntro = guardedNavigationAction<Context.Context>(
    () => {
      navigation.navigate(
        IdPayConfigurationRoutes.IDPAY_CONFIGURATION_NAVIGATOR,
        {
          screen: IdPayConfigurationRoutes.IDPAY_CONFIGURATION_INTRO,
          params: {}
        }
      );
    }
  );

  const navigateToIbanEnrollmentScreen = guardedNavigationAction(() =>
    navigation.navigate(
      IdPayConfigurationRoutes.IDPAY_CONFIGURATION_NAVIGATOR,
      {
        screen: IdPayConfigurationRoutes.IDPAY_CONFIGURATION_IBAN_ENROLLMENT,
        params: {}
      }
    )
  );

  const navigateToIbanOnboardingScreen = guardedNavigationAction(() =>
    navigation.navigate(
      IdPayConfigurationRoutes.IDPAY_CONFIGURATION_NAVIGATOR,
      {
        screen: IdPayConfigurationRoutes.IDPAY_CONFIGURATION_IBAN_LANDING
      }
    )
  );

  const navigateToIbanOnboardingFormScreen = guardedNavigationAction(() =>
    navigation.navigate(
      IdPayConfigurationRoutes.IDPAY_CONFIGURATION_NAVIGATOR,
      {
        screen: IdPayConfigurationRoutes.IDPAY_CONFIGURATION_IBAN_ONBOARDING
      }
    )
  );

  const navigateToInstrumentsEnrollmentScreen = guardedNavigationAction(() =>
    navigation.navigate(
      IdPayConfigurationRoutes.IDPAY_CONFIGURATION_NAVIGATOR,
      {
        screen:
          IdPayConfigurationRoutes.IDPAY_CONFIGURATION_INSTRUMENTS_ENROLLMENT,
        params: {}
      }
    )
  );

  const navigateToConfigurationSuccessScreen = () => {
    navigation.navigate(
      IdPayConfigurationRoutes.IDPAY_CONFIGURATION_NAVIGATOR,
      {
        screen: IdPayConfigurationRoutes.IDPAY_CONFIGURATION_SUCCESS
      }
    );
  };

  const navigateToInitiativeDetailScreen = (args: {
    context: Context.Context;
  }) => {
    const initiativeId = args.context.initiativeId;
    navigation.navigate(IDPayDetailsRoutes.IDPAY_DETAILS_MAIN, {
      screen: IDPayDetailsRoutes.IDPAY_DETAILS_MONITORING,
      params: { initiativeId }
    });
  };

  const showUpdateIbanToast = () => {
    IOToast.success(I18n.t(`idpay.configuration.iban.updateToast`));
  };

  const showFailureToast = (args: { context: Context.Context }) => {
    pipe(
      args.context.failure,
      InitiativeFailure.decode,
      O.fromEither,
      O.map(failure => I18n.t(`idpay.configuration.failureStates.${failure}`)),
      O.map(IOToast.error)
    );
  };

  const exitConfiguration = () => {
    navigation.pop();
  };

  return {
    navigateToConfigurationIntro,
    navigateToIbanEnrollmentScreen,
    navigateToIbanOnboardingScreen,
    navigateToIbanOnboardingFormScreen,
    navigateToInstrumentsEnrollmentScreen,
    navigateToConfigurationSuccessScreen,
    navigateToInitiativeDetailScreen,
    showUpdateIbanToast,
    showFailureToast,
    exitConfiguration
  };
};

export { createActionsImplementation };
