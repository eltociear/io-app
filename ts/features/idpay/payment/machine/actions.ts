import * as O from "fp-ts/lib/Option";
import { pipe } from "fp-ts/lib/function";
import I18n from "../../../../i18n";
import { useIONavigation } from "../../../../navigation/params/AppParamsList";
import { showToast } from "../../../../utils/showToast";
import { IDPayDetailsRoutes } from "../../details/navigation";
import { IdPayPaymentRoutes } from "../navigation/routes";
import { Context } from "./context";

const createActionsImplementation = (
  navigation: ReturnType<typeof useIONavigation>
) => {
  const navigateToAuthorizationScreen = () => {
    navigation.navigate(IdPayPaymentRoutes.IDPAY_PAYMENT_MAIN, {
      screen: IdPayPaymentRoutes.IDPAY_PAYMENT_AUTHORIZATION,
      params: {}
    });
  };

  const navigateToResultScreen = () =>
    navigation.navigate(IdPayPaymentRoutes.IDPAY_PAYMENT_MAIN, {
      screen: IdPayPaymentRoutes.IDPAY_PAYMENT_RESULT
    });

  const showErrorToast = () =>
    showToast(I18n.t("idpay.payment.authorization.error"), "danger", "top");

  const exitAuthorization = (context: Context) => {
    pipe(
      context.transactionData,
      O.map(({ initiativeId }) => {
        navigation.popToTop();
        navigation.navigate(IDPayDetailsRoutes.IDPAY_DETAILS_MAIN, {
          screen: IDPayDetailsRoutes.IDPAY_DETAILS_MONITORING,
          params: { initiativeId }
        });
      }),
      O.getOrElse(() => navigation.pop())
    );
  };

  return {
    navigateToAuthorizationScreen,
    navigateToResultScreen,
    showErrorToast,
    exitAuthorization
  };
};

export { createActionsImplementation };
