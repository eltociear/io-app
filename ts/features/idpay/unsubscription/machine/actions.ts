import { useIONavigation } from "../../../../navigation/params/AppParamsList";
import ROUTES from "../../../../navigation/routes";
import { IdPayUnsubscriptionRoutes } from "../navigation/routes";

const createActionsImplementation = (
  navigation: ReturnType<typeof useIONavigation>
) => {
  const navigateToConfirmationScreen = () => {
    navigation.navigate(
      IdPayUnsubscriptionRoutes.IDPAY_UNSUBSCRIPTION_NAVIGATOR,
      {
        screen: IdPayUnsubscriptionRoutes.IDPAY_UNSUBSCRIPTION_CONFIRMATION
      }
    );
  };

  const navigateToResultScreen = () =>
    navigation.navigate(
      IdPayUnsubscriptionRoutes.IDPAY_UNSUBSCRIPTION_NAVIGATOR,
      {
        screen: IdPayUnsubscriptionRoutes.IDPAY_UNSUBSCRIPTION_RESULT
      }
    );

  const exitUnsubscription = () => {
    navigation.pop();
  };

  const exitToWallet = () => {
    navigation.popToTop();
    navigation.navigate(ROUTES.MAIN, {
      screen: ROUTES.WALLET_HOME,
      params: { newMethodAdded: false }
    });
  };

  return {
    navigateToConfirmationScreen,
    navigateToResultScreen,
    exitUnsubscription,
    exitToWallet
  };
};

export { createActionsImplementation };
