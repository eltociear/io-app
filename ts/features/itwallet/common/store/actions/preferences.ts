import { ActionType, createStandardAction } from "typesafe-actions";

export const itwCloseFeedbackBanner = createStandardAction(
  "ITW_CLOSE_FEEDBACK_BANNER"
)<{ withFeedback?: boolean }>();

export type ItwPreferencesActions = ActionType<typeof itwCloseFeedbackBanner>;
