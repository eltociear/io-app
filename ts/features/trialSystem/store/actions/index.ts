import {
  ActionType,
  createAsyncAction,
  createStandardAction
} from "typesafe-actions";
import { TrialId } from "../../../../../definitions/trial_systwem/TrialId";
import { Subscription } from "../../../../../definitions/trial_systwem/Subscription";

type ErrorPayload = {
  trialId: TrialId;
  error: Error;
};

export const trialSystemStatusActivation = createAsyncAction(
  "TRIAL_SYSTEM_ACTIVATION_REQUEST",
  "TRIAL_SYSTEM_ACTIVATION_SUCCESS",
  "TRIAL_SYSTEM_ACTIVATION_FAILURE"
)<TrialId, Subscription, ErrorPayload>();

export const trialSystemStatusDeactivation = createAsyncAction(
  "TRIAL_SYSTEM_DEACTIVATION_REQUEST",
  "TRIAL_SYSTEM_DEACTIVATION_SUCCESS",
  "TRIAL_SYSTEM_DEACTIVATION_FAILURE"
)<TrialId, Subscription, ErrorPayload>();

export const trialSystemStatus = createAsyncAction(
  "TRIAL_SYSTEM_STATUS_REQUEST",
  "TRIAL_SYSTEM_STATUS_SUCCESS",
  "TRIAL_SYSTEM_STATUS_FAILURE"
)<TrialId, Subscription, ErrorPayload>();

export const trialSystemStatusReset = createStandardAction(
  "TRIAL_SYSTEM_STATUS_RESET"
)<TrialId>();

export type TrialSystemActions =
  | ActionType<typeof trialSystemStatus>
  | ActionType<typeof trialSystemStatusActivation>
  | ActionType<typeof trialSystemStatusReset>
  | ActionType<typeof trialSystemStatusDeactivation>;
