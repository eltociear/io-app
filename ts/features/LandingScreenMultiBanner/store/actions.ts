import { ActionType, createStandardAction } from "typesafe-actions";
import { LandingScreenBannerId } from "./reducer";

type UpdateBannerVisibilityPayloadType = {
  id: LandingScreenBannerId;
  enabled: boolean;
};
export const updateLandingScreenBannerVisibility = createStandardAction(
  "UPDATE_HOME_SCREEN_BANNER_STATE"
)<UpdateBannerVisibilityPayloadType>();

export type LandingScreenBannerActions = ActionType<
  typeof updateLandingScreenBannerVisibility
>;
