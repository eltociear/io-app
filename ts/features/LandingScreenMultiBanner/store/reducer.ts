import { getType } from "typesafe-actions";

import { Action } from "../../../store/actions/types";
import { updateLandingScreenBannerVisibility } from "./actions";

export const LANDING_SCREEN_BANNERS_ENABLED_MAP = {
  ITW_DISCOVERY: true,
  SETTINGS_DISCOVERY: true
} as const;

export type LandingScreenBannerId =
  keyof typeof LANDING_SCREEN_BANNERS_ENABLED_MAP;
export type LandingScreenBannerState = {
  [key in LandingScreenBannerId]: boolean;
};

export const landingScreenBannersReducer = (
  state: LandingScreenBannerState = LANDING_SCREEN_BANNERS_ENABLED_MAP,
  action: Action
) => {
  switch (action.type) {
    case getType(updateLandingScreenBannerVisibility):
      return {
        ...state,
        [action.payload.id]: action.payload.enabled
      };
  }
  return state;
};
