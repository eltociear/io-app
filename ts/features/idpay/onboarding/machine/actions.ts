import * as O from "fp-ts/lib/Option";
import { useIONavigation } from "../../../../navigation/params/AppParamsList";
import { useIODispatch } from "../../../../store/hooks";
import { guardedNavigationAction } from "../../../../xstate/helpers/guardedNavigationAction";
import { refreshSessionToken } from "../../../fastLogin/store/actions/tokenRefreshActions";
import { IDPayDetailsRoutes } from "../../details/navigation";
import { IdPayOnboardingRoutes } from "../navigation/routes";
import * as Context from "./context";

const createActionsImplementation = (
  navigation: ReturnType<typeof useIONavigation>,
  dispatch: ReturnType<typeof useIODispatch>
) => {
  const handleSessionExpired = () => {
    dispatch(
      refreshSessionToken.request({
        withUserInteraction: true,
        showIdentificationModalAtStartup: false,
        showLoader: true
      })
    );
  };

  const navigateToInitiativeDetailsScreen = guardedNavigationAction(() =>
    navigation.navigate(IdPayOnboardingRoutes.IDPAY_ONBOARDING_NAVIGATOR, {
      screen: IdPayOnboardingRoutes.IDPAY_ONBOARDING_INITIATIVE_DETAILS
    })
  );

  const navigateToPDNDCriteriaScreen = guardedNavigationAction(() =>
    navigation.navigate(IdPayOnboardingRoutes.IDPAY_ONBOARDING_NAVIGATOR, {
      screen: IdPayOnboardingRoutes.IDPAY_ONBOARDING_PDNDACCEPTANCE
    })
  );

  const navigateToBoolSelfDeclarationsScreen = guardedNavigationAction(() =>
    navigation.navigate(IdPayOnboardingRoutes.IDPAY_ONBOARDING_NAVIGATOR, {
      screen: IdPayOnboardingRoutes.IDPAY_ONBOARDING_BOOL_SELF_DECLARATIONS
    })
  );

  const navigateToMultiSelfDeclarationsScreen = guardedNavigationAction(
    (context: Context.Context) =>
      navigation.navigate({
        name: IdPayOnboardingRoutes.IDPAY_ONBOARDING_NAVIGATOR,
        params: {
          screen: IdPayOnboardingRoutes.IDPAY_ONBOARDING_MULTI_SELF_DECLARATIONS
        },
        key: String(context.multiConsentsPage)
      })
  );

  const navigateToCompletionScreen = () =>
    navigation.navigate(IdPayOnboardingRoutes.IDPAY_ONBOARDING_NAVIGATOR, {
      screen: IdPayOnboardingRoutes.IDPAY_ONBOARDING_COMPLETION
    });

  const navigateToFailureScreen = () =>
    navigation.navigate(IdPayOnboardingRoutes.IDPAY_ONBOARDING_NAVIGATOR, {
      screen: IdPayOnboardingRoutes.IDPAY_ONBOARDING_FAILURE
    });

  const navigateToInitiativeMonitoringScreen = (context: Context.Context) => {
    if (O.isNone(context.initiative)) {
      throw new Error("Initiative is undefined");
    }

    navigation.replace(IDPayDetailsRoutes.IDPAY_DETAILS_MAIN, {
      screen: IDPayDetailsRoutes.IDPAY_DETAILS_MONITORING,
      params: {
        initiativeId: context.initiative.value.initiativeId
      }
    });
  };

  const closeOnboarding = () => {
    navigation.popToTop();
  };

  return {
    handleSessionExpired,
    navigateToInitiativeDetailsScreen,
    navigateToPDNDCriteriaScreen,
    navigateToBoolSelfDeclarationsScreen,
    navigateToMultiSelfDeclarationsScreen,
    navigateToCompletionScreen,
    navigateToFailureScreen,
    navigateToInitiativeMonitoringScreen,
    closeOnboarding
  };
};

export { createActionsImplementation };
