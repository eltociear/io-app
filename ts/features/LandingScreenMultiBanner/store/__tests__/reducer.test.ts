import { applicationChangeState } from "../../../../store/actions/application";
import { landingScreenBannersReducer } from "../reducer";

describe("landingScreenBannersReducer", () => {
  it("should match snapshot", () => {
    expect(
      landingScreenBannersReducer(undefined, applicationChangeState("active"))
    ).toMatchSnapshot();
  });
});
