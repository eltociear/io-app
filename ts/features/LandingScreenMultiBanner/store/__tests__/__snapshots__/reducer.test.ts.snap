// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landingScreenBannersReducer should match snapshot 1`] = `
{
  "ITW_DISCOVERY": true,
  "SETTINGS_DISCOVERY": true,
}
`;

